core/src/test/scala/unit/kafka/controller/ControllerIntegrationTest.scala

```diff
@@ -490,6 +490,41 @@ class ControllerIntegrationTest extends QuorumTestHarness {
       "failed to get expected partition state upon broker startup")
   }
 
+  @Test
+  def testAutoPreferredReplicaLeaderElectionWithReassigningPartitions(): Unit = {
```

Comment:
I think it would still work fine, but did we consider a situation where the preferred leader election occurs right before or at the same time as the reassignment completes? 
For example, we have a reassignment going, a broker is restarted, and we may or may not have preferred leader election before it completes.

Reply:
During a partition's reassignment, if the current leader is not in TRS or isn't alive, a leader election will be triggered at the time when the TRS = ISR. At that time, the partition is still in `controllerContext.partitionsBeingReassigned`, so `checkAndTriggerAutoLeaderRebalance` will skip that particular partition. 
When the reassignment is done (partition being removed from `controllerContext.partitionsBeingReassigned`),  `checkAndTriggerAutoLeaderRebalance` will treat it as a normal partition, preferred leader election may be triggered if it's allowed to.